src/components/applicationForm/ApplicationFormActions.tsx

```diff
@@ -1,150 +1,155 @@
-            <p className="text-center text-gray-500">
-              {applicationFormStatusData[displayStatus].description}
-            </p>
-
-            {data.status !== UserStatus.OnWaitList && (
-              <Button
-                className="bg-primary-dark text-white px-6 py-3 rounded-md"
-                onClick={handleClick}
-              >
-                {applicationFormStatusData[displayStatus].buttonText}
-              </Button>
-            )}
-
-            <div className="text-sm text-center text-gray-600">
-              Learn more about The Gym software developer trainee program{" "}
-              <Link
-                to="https://www.the-gym.rw/"
-                className="text-blue-600 underline"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                here
-              </Link>
-              .
-            </div>
-          </div>
-        </>
-      )}
-    </>
-  )
-}
+import { useContext, useEffect, useState } from "react"
+import { useDispatch } from "react-redux"
+import { Link } from "react-router-dom"
+import { showAlert } from "../../features/user/alertSlice"
+import { useGetProfileQuery } from "../../features/user/backendApi"
+import { applicationFormStatusData } from "../../utils/data"
+import { getApplicationFormStatus, getFormattedDate } from "../../utils/helper"
+import {
+  AlertType,
+  ApplicationForm,
+  ApplicationFormStatus,
+  ButtonSize,
+  UserResponseQuestion,
+  UserRole,
+  UserStatus,
+} from "../../utils/types"
+import Button from "../ui/Button"
+import CohortInfo from "../ui/CohortInfo"
+import { SocketContext } from "../../utils/contexts/SocketContext"
+
+interface ApplicationFormActionsProps {
+  applicationForm: Omit<ApplicationForm, "questions"> & {
+    questions: UserResponseQuestion[]
+    trainingStartDate: string
+  }
+  role: UserRole
+  userStatus: UserStatus
+}
+
+export default function ApplicationFormActions({
+  applicationForm,
+  role,
+  userStatus,
+}: ApplicationFormActionsProps) {
+  const status =
+    role === UserRole.Prospect && userStatus === UserStatus.Applied
+      ? ApplicationFormStatus.Submitted
+      : getApplicationFormStatus(applicationForm)
+
+  const { data, refetch, isLoading } = useGetProfileQuery()
+  const dispatch = useDispatch()
+
+  const [displayStatus, setDisplayStatus] = useState<ApplicationFormStatus>(
+    () =>
+      data?.status === UserStatus.OnWaitList
+        ? ApplicationFormStatus.JoinedWaitList
+        : status,
+  )
+
+  const { socket } = useContext(SocketContext)
+
+  const handleClick = () => {
+    window.open(applicationFormStatusData[displayStatus].buttonLink, "_blank")
+  }
+
+  useEffect(() => {
+    if (socket && data?.email) {
+      socket.emit("join-room", data.email)
+
+      socket.on("joinedTheWaitList", (message) => {

```

Comment:
follow the same naming convention for the events. It should be in `kebab-case`. I would suggest to also create an enum for the socket events.
eg:
```ts
enum SocketEvent {
   JoinRoom = 'join-room',
   JoinedWaitlist = 'joined-waitlist',
   WaitlistError = 'waitlist-error'
}
```